io.c

```diff
@@ -10056,45 +10059,32 @@ advice_arg_check(VALUE advice)
 
 /*
  *  call-seq:
- *     ios.advise(advice, offset=0, len=0) -> nil
- *
- *  Announce an intention to access data from the current file in a
- *  specific pattern. On platforms that do not support the
- *  <em>posix_fadvise(2)</em> system call, this method is a no-op.
+ *    advise(advice, offset = 0, len = 0) -> nil
  *
- *  _advice_ is one of the following symbols:
+ *  Announces an intention to access data from the current file in a specific pattern.
  *
- *  :normal::     No advice to give; the default assumption for an open file.
- *  :sequential:: The data will be accessed sequentially
- *                with lower offsets read before higher ones.
- *  :random::     The data will be accessed in random order.
- *  :willneed::   The data will be accessed in the near future.
- *  :dontneed::   The data will not be accessed in the near future.
- *  :noreuse::    The data will only be accessed once.
+ *  Has no effect  on a platform that does not support system call
+ *  {posix_fadvise(2)}[https://linux.die.net/man/2/posix_fadvise].
```

Comment:
I have a thought on this group of methods (more or less directly mapped into low-level POSIX call)... That to lessen the confusion, we might structure them this way:
> Does POSIX call `foo` (links to docs), which (short informal explanation)

Because the docs starting with "Announces an intention..." leaves—especially to novice—unlear the possible effect: announces to whom? Maybe it is some cool Ruby tricky I need to know? While starting with (intentionally boring) "does this system thing" makes it clear "you'll know when you need it; not now."

WDYT?

Reply:
For `advise`, this comment seems not to be addressed? :thinking:

Reply:
Modified.

Reply:
Modified.

Reply:
> I have a thought on this group of methods (more or less directly mapped into low-level POSIX call)... That to lessen the confusion, we might structure them this way:
> 
> > Does POSIX call `foo` (links to docs), which (short informal explanation)
> 

This is one of the five Posix calls.  I thought you meant to have a short informal explanation for each.

Reply:
No, sorry for misunderstanding. What I meant then, was that:
1. For POSIX calls, it is important to start the explanation by pointing out that it is a system call, not just some Ruby idiom
2. ...and maybe not repeat the generic information of what this call does in too many words (I mean, not go into explaining POSIX system calls concepts in general, just duplicating the `man` page).

I have nothing against good explanation in context of Ruby, and definitely for more "core" methods like `pipe`, the examples and details are helpful. It is rather more esoteric stuff like `advise` don't need too much details (because _if_ you need that, you should know _why_), but `pipe` explanations and examples are perfectly fine by me.

Please excuse me for confusing comments :(